runtime/service/src/main/java/org/apache/polaris/service/quarkus/config/QuarkusProducers.java

```diff
@@ -170,6 +175,34 @@ public UserSecretsManagerFactory userSecretsManagerFactory(
     return userSecretsManagerFactories.select(Identifier.Literal.of(config.type())).get();
   }
 
+  @Produces
+  @Singleton
+  @Identifier("http-client-s3")
+  public SdkHttpClient sdkHttpClient(S3AccessConfig config) {
+    ApacheHttpClient.Builder httpClient = ApacheHttpClient.builder();
+    config.maxHttpConnections().ifPresent(httpClient::maxConnections);
+    config.readTimeout().ifPresent(httpClient::socketTimeout);
+    config.connectTimeout().ifPresent(httpClient::connectionTimeout);
+    config.connectionAcquisitionTimeout().ifPresent(httpClient::connectionAcquisitionTimeout);
+    config.connectionMaxIdleTime().ifPresent(httpClient::connectionMaxIdleTime);
+    config.connectionTimeToLive().ifPresent(httpClient::connectionTimeToLive);
+    config.expectContinueEnabled().ifPresent(httpClient::expectContinueEnabled);
+    return httpClient.build();
+  }
+
+  public void closeSdkHttpClient(@Disposes @Identifier("http-client-s3") SdkHttpClient client) {
```

Comment:
This feels out of place in `QuarkusProducers`. Maybe we could create a wrapper over `SdkHttpClient` and handle that like our other `Closeable`s or use `AutoCloseable`

Reply:
What do you mean by "out of place"? `QuarkusProducers` creates a singleton client, `QuarkusProducers` also closes that same client. Who would call `.close()` on the wrapper?